adapters/nexx360/nexx360.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+	"github.com/prebid/prebid-server/v3/version"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type Ext struct {
+	Nexx360 json.RawMessage `json:"nexx360"`
+}
+
+type Nexx360Caller struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+type ReqExt struct {
+	Nexx360 *ReqNexx360Ext `json:"nexx360,omitempty"`
+}
+
+type ReqNexx360Ext struct {
+	Caller []Nexx360Caller `json:"caller,omitempty"`
+}
+
+type Nexx360ResBidExt struct {
+	BidType string `json:"bidType,omitempty"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{endpoint: config.Endpoint}, nil
+}
+
+// CALLER Info used to track Prebid Server
+// as one of the hops in the request to exchange
+
+func getVersion() string {
+	if version.Ver != "" {
+		return version.Ver
+	}
+	return "n/a"
+}
+
+var CALLER = Nexx360Caller{"Prebid-Server", getVersion()}
+
+func processImps(impList []openrtb2.Imp) (imp []openrtb2.Imp, tagId string, placement string, error error) {
+	var imps []openrtb2.Imp
+	for idx, imp := range impList {
+		var bidderExt adapters.ExtImpBidder
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return nil, "", "", &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		var nexx360Ext openrtb_ext.ExtImpNexx360
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &nexx360Ext); err != nil {
+			return nil, "", "", &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		impExt := Ext{
+			Nexx360: bidderExt.Bidder,
+		}
+
+		impExtJSON, err := json.Marshal(impExt)
+		if err != nil {
+			return nil, "", "", &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		imp.Ext = impExtJSON
+		imps = append(imps, imp)
+		if idx == 0 {
+			var nexx360Ext openrtb_ext.ExtImpNexx360
+			if err := jsonutil.Unmarshal(bidderExt.Bidder, &nexx360Ext); err != nil {
+				return nil, "", "", &errortypes.BadInput{
+					Message: err.Error(),
+				}
+			}
+			tagId = nexx360Ext.TagId
+			placement = nexx360Ext.Placement
+		}
+
+	}
+
+	return imps, tagId, placement, nil
+}
+
+func makeReqExt() ([]byte, error) {
+	reqExt := ReqExt{
+		Nexx360: &ReqNexx360Ext{
+			Caller: []Nexx360Caller{CALLER},
+		},
+	}
+
+	return json.Marshal(reqExt)
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var imp, tagId, placement, err = processImps(request.Imp)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	request.Imp = imp
+
+	urlBuilder, err := url.Parse(a.endpoint)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	query := url.Values{}
+
+	const placementKey = "placement"
+	if placement != "" {
+		query[placementKey] = []string{placement}
```

Comment:
I'm not sure why you're appending a slice of strings. I think it would be better to use `query.Add` and get rid of the const: `query.Add("placement", placement)`.

Reply:
Done